cql3/type_json.cc

```diff
@@ -194,7 +194,16 @@ struct from_json_object_visitor {
         }
         return t.decompose(T(to_int64_t(value)));
     }
-    bytes operator()(const string_type_impl& t) { return t.from_string(rjson::to_string_view(value)); }
+    bytes operator()(const string_type_impl& t) {
+        if (value.IsString()) {
+            return t.from_string(rjson::to_string_view(value)); 
+        } else if (value.IsBool() || value.IsNumber()) {
+            return t.from_string(rjson::print(value));
```

Comment:
How does rjson::print() print floating point numbers? Does it have the full accuracy?
If we accept this patch, we should have a test for this.

Reply:
I saw later that you do have a test for this (with some strange results showing slightly different last digit in the two versions, but I guess it's meaningless).

Reply:
RapidJSON uses its own floating-point printing code (https://github.com/Tencent/rapidjson/blob/e4bde977440d4a00f820b6586899e48a972d2493/include/rapidjson/internal/dtoa.h). By default it seems to have quite a long maximum precision configured: `static const int kDefaultMaxDecimalPlaces = 324;`, however it does use scientific notation for large numbers.

In the testing code, I compared the output of it to the previous Scylla 4.0.0 code with JsonCpp. It looks very similar, but for some cases, it printed numbers less precise compared to JsonCpp (despite the same number of digits).

Here is discussion with @psarna about using `rjson::print` instead of manually printing floats: https://github.com/scylladb/scylla/pull/10101#discussion_r810143380

Reply:
@nyh "Does it have the full accuracy? If we accept this patch, we should have a test for this." What type of test would you add to the existing tests? Some large number? (i.e. `1e200`?)

Reply:
No, accuracy is not about the range of exponent (like 200 in your example) but the digits of precision.
For floating-point the accuracy is less important (because I don't think we make any guarantees about it...) but for integers, we can't loose any digit of precision. I gave in a comment below the case of the highest bigint - 9223372036854775808, which has (one) too many digits of accuracy for a double, so if we write 9223372036854775808E0 and set it into a bigint, we can find ourselves with a slightly different integer, like 9223372036854775807. I suggested below we should have a test for this (even if it will still fail in the current implementation).